src/types.ts

```diff
@@ -0,0 +1,8 @@
+import { ExitCodes } from "./constants";
+
+export type PotentialPromise<T> = Promise<T> | T;
```

Comment:
TypeScript has a `PromiseLike` type for this IIRC.
```suggestion
export type PotentialPromise<T> = PromiseLike<T>;
```